app/assets/javascripts/components/Cart.jsx

```diff
@@ -0,0 +1,61 @@
+var Cart = React.createClass({
+  getInitialState: function(){
+     return {
+       initialCP: this.props.carted_products,
+       initialPT: this.props.products_total,
+       initialCS: this.props.carted_subscriptions,
+       initialCT: this.props.cart_total
+     }
+  },
+  componentWillMount: function(){
+    this.setState({
+      carted_products: this.state.initialCP,
+      products_total: this.state.initialPT,
+      carted_subscriptions: this.state.initialCS,
+      cart_total: this.state.initialCT
+    })
+  },
+  handleProductsEmpty: function() {
+    this.setState({carted_products: []});
+  },
+  handleSubscriptionsEmpty: function() {
+    this.setState({carted_subscriptions: []});
+  },
+  _renderCartedProducts: function(){
+    if(this.state.carted_products.length !== 0) {
```

Comment:
Hmm...interesting tangent., it used to be that doing 'if (!array.length) {...}` would be faster check and considered better practice. - but you can test here: [array.length](https://jsperf.com/test-of-array-length). At least in Chrome v60, ` if (array.length > 0)` is faster now.